docs/style-guides/js.md

```diff
@@ -0,0 +1,252 @@
+# origin-web-console JS style guide
+
+A reasonable approach to assisting in style consistency in `openshift-web-console`
+inspired by guides such as the style guide published by [airbnb](https://github.com/airbnb/javascript).
+
+## ESLint
+
+Run `grunt eslint` to check code for errors and style discrepancies.
+
+## Examples
+
+
+### Variables
+
+Group your variable names but use individual `var` statements:
```

Comment:
Heh, I think right now we really suffer from having no structure around this.  I'd prob like to suggest we pick something going forward & update old code to match when possible.

For the sake of furthering discussion:

@spadget  I think the "declare a variable near where it is used" might mean that entire block could be pulled out into a separate function.  Big functions seem to get harder and harder to maintain, so I definitely prefer asserting rules that help us work towards smaller fns, even if we cant perfectly adhere to it.

@jwforres I actually prefer that as well & did it in my early PRs here, but I think I was asked to turn my vars into individual var statements so I've done this ever since. :smile: